db/classManagement.go

```diff
@@ -0,0 +1,305 @@
+		http.Error(w, "Error occurred in reading body.", http.StatusInternalServerError)
+		return
+	}
+
+	if req.UID == "" {
+		http.Error(w, "Error occurred in reading UID.", http.StatusInternalServerError)
+		return
+	}
+	
+	if req.Name == "" {
+		http.Error(w, "Error occurred in reading Name.", http.StatusInternalServerError)
+		return
+	}
+
+	if req.Thumbnail < 0 || req.Thumbnail >= 50 {
+		http.Error(w, "Bad thumbnail provided, Exiting", http.StatusInternalServerError)
+		return
+	}
+
+	
+
+	// structure for class info
+	class := Class{
+		Thumbnail: req.Thumbnail, 
+		Name: req.Name, 
+		Creator: req.UID, 
+		Instructors: []string{req.UID},
+		Members: []string{},
+		Programs: []string{},
+	}	
+
+	
+
+	// create the class
+	cid, err := d.CreateClass(r.Context(), &class)
+	if err != nil {
+		http.Error(w, "Error updating class in Firebase", http.StatusInternalServerError)
+		return
+	}
+
+	//create an wid for this class
+	wid, err := d.MakeAlias(r.Context(), cid, ClassesAliasPath)
+
+	//Update class info 
+	// create the class
+	err = d.UpdateClassWID(r.Context(), cid, wid)
+
+	//add this class to the user's "Classes" list
+	err = d.AddClassToUser(r.Context(), req.UID, cid)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// read the class document just created 
+	c, err := d.GetClass(r.Context(), cid)
+	if err != nil || c == nil {
+		http.Error(w, "Class does not exist", http.StatusNotFound)
+		return
+	}
+	//return the class struct in the response
+	if resp, err := json.Marshal(c); err != nil {
+		http.Error(w, "Failed to marshal response", http.StatusInternalServerError)
+	} else {
+		w.Write(resp)
+	}
+}
+
+// HandleGetClass takes the UID (either of a member or an instructor) 
+// and a CID (wid) as a JSON, and returns an object representing the class. 
+// If the given UID is not a member or an instructor, error is returned
+func (d *DB) HandleGetClass(w http.ResponseWriter, r *http.Request) {
+	
+	var err error
+
+	//create an anonymous structure to handle requests
+	req := struct {
+		UID 		string  	`json:"uid"`
+		WID 		string		`json:"cid"`
+	}{}
+
+	//read JSON from request body
+	if err = requests.BodyTo(r, &req); err != nil {
+		http.Error(w, "Error occurred in reading body", http.StatusInternalServerError)
+		return
+	}
+	if req.UID == "" {
+		http.Error(w, "Error occurred in reading uid", http.StatusInternalServerError)
+		return
+	}
+	if req.WID == "" {
+		http.Error(w, "Error occurred in reading cid", http.StatusInternalServerError)
+		return
+	}
+
+	cid, err := d.GetUIDFromWID(r.Context(), req.WID, ClassesAliasPath)
+
+	// get the class as a struct (pointer)
+	c, err := d.GetClass(r.Context(), cid)
+
+	// check for error
+	if err != nil || c == nil {
+		http.Error(w, "Failed to get class (class does not exist or failed to unmarshal data)", http.StatusNotFound)
+		return
+	}
+
+	//check if the uid exists in the members list or instructor list
+	var is_in bool = false;
```

Comment:
I noticed that you are slipping into a more JavaScript-y style here. Remember that you can always declare it as `is_in := false`